integration-tests/api/__tests__/store/product-type.js

```diff
@@ -0,0 +1,160 @@
+const path = require("path")
+
+const { IdMap } = require("medusa-test-utils")
+
+const setupServer = require("../../../helpers/setup-server")
+const { useApi } = require("../../../helpers/use-api")
+const { initDb, useDb } = require("../../../helpers/use-db")
+
+const productSeeder = require("../../helpers/product-seeder")
+const {
+  DiscountRuleType,
+  AllocationType,
+  DiscountConditionType,
+  DiscountConditionOperator,
+} = require("@medusajs/medusa")
+const { simpleDiscountFactory } = require("../../factories")
+
+jest.setTimeout(50000)
+
+describe("/store/product-types", () => {
+  let medusaProcess
+  let dbConnection
+
+  beforeAll(async () => {
+    const cwd = path.resolve(path.join(__dirname, "..", ".."))
+    dbConnection = await initDb({ cwd })
+    medusaProcess = await setupServer({ cwd })
+  })
+
+  afterAll(async () => {
+    const db = useDb()
+    await db.shutdown()
+
+    medusaProcess.kill()
+  })
+
+  describe("GET /store/product-types", () => {
+    beforeEach(async () => {
+      await productSeeder(dbConnection)
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("returns a list of product types", async () => {
+      const api = useApi()
+
+      const res = await api.get("/store/product-types").catch((err) => {
+        console.log(err)
+      })
+
+      expect(res.status).toEqual(200)
+
+      const typeMatch = {
+        created_at: expect.any(String),
+        updated_at: expect.any(String),
+      }
+
+      expect(res.data.product_types).toMatchSnapshot([typeMatch, typeMatch])
```

Comment:
I'll skip this one for now. I have yet to wrap my head on how I would like to move away from snapshots.

Reply:
It is just for consistency as most a the new tests are tested that way. Also, here what we really want to test is if the expected types are in the list (check id) and if the number of types is the one expected (here 2)

Reply:
These where copied from the admin endpoint. I still stand by my decision to not refactor those tests.

In the new suggested format, I don't like that we assume that the seeded entities have id "XYZ". I rather be able to reference their value instead of assuming it explicitly. Something along the lines of:
```
import { mockProductTypes } from "product-seeded"
expect(res.data.product_types).toEqual(
  expect.arrayContainer([
    expect.objectContaining({
      id: mockProductTypes[0].id
    }),
    expect.objectContaining({
      id: mockProductTypes[1].id
    }),
  ])
)
```
I don't not want to further expand the scope of this PR.

Reply:
Sounds perfect, I agree with you